ocaml/typing/types.mli

```diff
@@ -532,6 +532,9 @@ type type_declaration =
     type_unboxed_default: bool;
     (* true if the unboxed-ness of this type was chosen by a compiler flag *)
     type_uid: Uid.t;
+    type_has_illegal_crossings: bool;
```

Comment:
I see why this is here, but it should get removed once we remove illegal crossings. I think it's worth leaving a CR to do so.

Actually, I bet we'll be able to kill it once we can merge the improvements to printing. If not, let's circle back.